synapse/push/bulk_push_rule_evaluator.py

```diff
@@ -125,28 +109,43 @@ def __init__(self, hs: "HomeServer"):
         # Whether to support MSC3772 is supported.
         self._relations_match_enabled = self.hs.config.experimental.msc3772_enabled
 
+    async def get_rules(
+        self, room_id: str, event: EventBase
+    ) -> Dict[str, List[Dict[str, dict]]]:
+        """Given an event return the rules for all users who are
+        currently in the room.
+        """
+
+        local_users = await self.store.get_local_users_in_room(room_id)
```

Comment:
Insight from E:

"Find out who is in the room NOW, and then what they can see" is cheaper than "find out who was in the room when the event was received" because

- who is in the room now is immediately available and doesn't require fetching state
- there are probably relatively few local users compared to loads of remote users in a large room.

Might be nice to have in a docstring or comment somewhere.